File: tanks_backend/src/main/java/hr/antitalent/tanks_backend/websocket/game/PlayerGameInfo.java
package hr.antitalent.tanks_backend.websocket.game;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PlayerGameInfo {
    // Player info 1
    private String playerId;
    
    // Position and movement
    private double x;
    private double y;
    private double rotation; 
    private double turretRotation;
    private double velocity;
    private boolean isMoving;

    // Tank state
    private int health;
}

File: TanksGame/Assets/Tanks/Game Scripts/MPGame/PlayerGameInfo.cs
[System.Serializable]
public class PlayerGameInfo
{
    // Player info
    public long playerId;
    
    // Position and movement
    public float x;
    public float y;
    public float rotation;
    public float turretRotation;
    public float velocity;
    public bool isMoving;
    
    // Tank state
    public float health;

    public PlayerGameInfo() { }

    public PlayerGameInfo(
        long playerId, 
        float x, float y, float rotation, float turretRotation, float velocity, 
        float health, 
        bool isMoving)
    {
        this.playerId = playerId;
        this.x = x; this.y = y;
        this.rotation = rotation; 
        this.turretRotation = turretRotation; 
        this.velocity = velocity; 
        this.isMoving = isMoving;
        this.health = health;
    }
}